controllers/openstackclient_controller.go

```diff
@@ -441,122 +496,85 @@ func (r *OpenStackClientReconciler) podCreateOrUpdate(
 	}
 	annotation := fmt.Sprintf("[%s]", annotations)
 
-	env := common.MergeEnvs([]corev1.EnvVar{}, *envVars)
-
+	// Create a (mostly) empty pod spec
 	pod := &corev1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      instance.Name,
-			Namespace: instance.Namespace,
-			Annotations: map[string]string{
-				"k8s.v1.cni.cncf.io/networks": annotation,
-			},
+			Name:        instance.Name,
+			Namespace:   instance.Namespace,
+			Annotations: map[string]string{},
 		},
 	}
-	pod.Spec = corev1.PodSpec{
-		ServiceAccountName: openstackclient.ServiceAccount,
-		SecurityContext: &corev1.PodSecurityContext{
-			RunAsUser:  &runAsUser,
-			RunAsGroup: &runAsGroup,
-			FSGroup:    &runAsGroup,
-		},
-		TerminationGracePeriodSeconds: &terminationGracePeriodSeconds,
-		Volumes:                       volumes,
-		Containers: []corev1.Container{
-			{
-				Name:            "openstackclient",
-				Image:           instance.Spec.ImageURL,
-				ImagePullPolicy: corev1.PullAlways,
-				Env:             env,
-				VolumeMounts:    volumeMounts,
-			},
+	common.InitMap(&pod.Labels)
+	pod.Spec = corev1.PodSpec{}
+	pod.Spec.SecurityContext = &corev1.PodSecurityContext{}
+	pod.Spec.DNSConfig = &corev1.PodDNSConfig{}
+	pod.Spec.Containers = []corev1.Container{
+		{
+			Name: "openstackclient",
```

Comment:
this could probably be instance.name?

Reply:
Could I guess  (I've not change it in this patch). Does it make much difference in practice?

Reply:
no it does not make a different, and thinking about it again it might be even better to have it fixed, like when you create a tempest openstackclient, the container will sill be openstackclient, instead of tempest. probably easier for debugging